SpatialGDK/Source/Private/Interop/SpatialPlayerSpawner.cpp

```diff
@@ -51,7 +52,39 @@ void USpatialPlayerSpawner::SendPlayerSpawnRequest()
 	Schema_Object* RequestObject = Schema_GetCommandRequestObject(CommandRequest.schema_type);
 	AddStringToSchema(RequestObject, 1, DummyURL.ToString(true));
 
-	NetDriver->Connection->SendCommandRequest(SpatialConstants::SPAWNER_ENTITY_ID, &CommandRequest, 1);
+	// Send an entity query for the SpatialSpawner and bind a delegate so that once it's found, we send a spawn command.
+	Worker_Constraint SpatialSpawnerConstraint;
+	SpatialSpawnerConstraint.constraint_type = WORKER_CONSTRAINT_TYPE_COMPONENT;
+	SpatialSpawnerConstraint.component_constraint.component_id = SpatialConstants::PLAYER_SPAWNER_COMPONENT_ID;
+
+	Worker_EntityQuery SpatialSpawnerQuery{};
+	SpatialSpawnerQuery.constraint = SpatialSpawnerConstraint;
+	SpatialSpawnerQuery.result_type = WORKER_RESULT_TYPE_SNAPSHOT;
+
+	Worker_RequestId RequestID;
+	RequestID = NetDriver->Connection->SendEntityQueryRequest(&SpatialSpawnerQuery);
+
+	EntityQueryDelegate SpatialSpawnerQueryDelegate;
+	SpatialSpawnerQueryDelegate.BindLambda([this, RequestID, CommandRequest](Worker_EntityQueryResponseOp& Op)
+	{
+		if (Op.status_code != WORKER_STATUS_CODE_SUCCESS)
+		{
+			UE_LOG(LogSpatialPlayerSpawner, Error, TEXT("Entity query for SpatialSpawner failed: %s"), UTF8_TO_TCHAR(Op.message));
+		}
+		else if (Op.result_count == 0)
+		{
+			UE_LOG(LogSpatialPlayerSpawner, Error, TEXT("Could not find SpatialSpawner via entity query: %s"), UTF8_TO_TCHAR(Op.message));
+		}
+		else
+		{
+			checkf(Op.result_count == 1, TEXT("There should never be more than one SpatialSpawner entity."));
+
+			NetDriver->Connection->SendCommandRequest(Op.results[0].entity_id, &CommandRequest, 1);
```

Comment:
Why don't we create the player spawn request here, just before sending it?

Reply:
Hmmmmm, good point.